app/controllers/urls_controller.rb

```diff
@@ -0,0 +1,49 @@
+class UrlsController < ApplicationController
+  before_action  :set_url, only: [:show, :update]
+
+  def index
+    @urls = Url.order("pinned DESC, created_at DESC")
+  end
+
+  def create
+    @url = Url.new(url_params)
+    @url.assign_random_string_for_slug
+    if @url.save
+      render status: :ok, json: {notice: "Success"}
+    else
+      render status: :unprocessable_entity, json: {errors: @url.errors.full_messages}
+    end
+  end
+  
+  def show
+    @url.clicked +=1
```

Comment:
The show action is only to fetch a model from the database. It shouldn't be used to update a model. Updating the model should happen in the update action.